src/hb-ot-metrics.cc

```diff
@@ -69,9 +69,20 @@ hb_ot_metrics_get (hb_font_t       *font,
 #define GET_X_METRIC(TABLE, ATTR) GET_METRIC (TABLE, ATTR, x)
 #define GET_Y_METRIC(TABLE, ATTR) GET_METRIC (TABLE, ATTR, y)
 
-  case HB_OT_METRICS_HORIZONTAL_ASCENDER:         return GET_Y_METRIC (hmtx, ascender);
-  case HB_OT_METRICS_HORIZONTAL_DESCENDER:        return GET_Y_METRIC (hmtx, descender);
-  case HB_OT_METRICS_HORIZONTAL_LINE_GAP:         return GET_Y_METRIC (hmtx, line_gap);
+#define USE_TYPO_METRICS (1u<<7)
+  case HB_OT_METRICS_HORIZONTAL_ASCENDER:
+    return ((font->face->table.OS2->fsSelection & USE_TYPO_METRICS == 0) &&
+	    GET_Y_METRIC (hhea, ascender)) ||
```

Comment:
It would read more natural to test for `^ USE_TYPO_METRICS) && OS2...) || hhead...`.

Reply:
Done